File: leetcode-new/53.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        int curr = 0;
        int ret = INT_MIN;
        for (int i = 0; i < n; ++i)
        {
            curr = max(curr + nums[i], nums[i]);
            ret = max(ret, curr);
        }
        return ret;
    }
};

File: leetcode-larry/53.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        gmax = -inf
        lmax = 0

        for n in nums:
            lmax += n
            gmax = max(gmax, lmax)
            if lmax < 0:
                lmax = 0
        
        return gmax
